pgxn/neon/hll.c

```diff
@@ -127,30 +126,25 @@ addSHLL(HyperLogLogState *cState, uint32 hash)
 	/* Compute the rank of the remaining 32 - "k" (registerWidth) bits */
 	count = rho(hash << HLL_BIT_WIDTH, HLL_C_BITS);
 
-	for (i = 0, j = 0; i < cState->regs[index].size; i++)
+	for (i = 0; i < count + 1; i++)
 	{
-		if (cState->regs[index].fpm[i].ts >= now - cState->window
-			&& cState->regs[index].fpm[i].R > count)
-		{
-			cState->regs[index].fpm[j++] = cState->regs[index].fpm[i];
-		}
+		cState->regs[index][i] = now;
```

Comment:
I wonder why do we need tp assign current timestamp to all elements with index less or equal then count?
Why it is not enough just to assign:
```
cState->regs[index][count] = now;
```

Reply:
Sorry, I do not understand it.
We need to find maximal count, right?
So your loop:
```
for (size_t i = 0; i < HLL_C_BITS + 1; i++)
	{
		if (reg[i] >= since)
		{
			max = i;
		}
	}
```
is actually locating the largest R with timestamp within specified period.

Reply:
1. Binary search in array of size 22 seems to have no sense.
2. Time of calculating estimation is not critical: it is needed rarely when autoscaler agent needs to update its working set estimation: once per second or even per minute.
3. Time of updating hashes is much more critical because it is called on each SMGR read operation

So we should optimise `addHLL` rather than `getMaximum`